### PR TITLE
Implement translation of Snowflake UDF creation commands

### PR DESCRIPTION
**Pull request by vil1**

### CODE REVIEW

File: core/src/test/scala/com/databricks/labs/remorph/parsers/ParserTestCommon.scala [hunk 1]
```diff
@@ -2,10 +2,10 @@ package com.databricks.labs.remorph.parsers
 
 import com.databricks.labs.remorph.parsers.intermediate.{NamedTable, Relation, TreeNode}
 import org.antlr.v4.runtime.tree.ParseTreeVisitor
-import org.antlr.v4.runtime.{CharStream, CharStreams, CommonTokenStream, RuleContext, TokenSource, TokenStream}
+import org.antlr.v4.runtime.{CharStream, CharStreams, CommonTokenStream, Parser, RuleContext, TokenSource, TokenStream}
 import org.scalatest.{Assertion, Assertions}
 
-trait ParserTestCommon[P] { self: Assertions =>
+trait ParserTestCommon[P <: Parser] { self: Assertions =>
```
Comment (vil1) [inline core/src/test/scala/com/databricks/labs/remorph/parsers/ParserTestCommon.scala:8]: This is needed when one uncomments the `println` line bellow.

File: core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/DataTypeBuilder.scala [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import com.databricks.labs.remorph.parsers.snowflake.SnowflakeParser.Data_typeContext
+import com.databricks.labs.remorph.parsers.{intermediate => ir}
+
+import scala.collection.JavaConverters._
+object DataTypeBuilder {
+
+  def buildDataType(ctx: Data_typeContext): ir.DataType = {
+    val sizeOpt = Option(ctx.data_type_size()).map(_.getText.toInt)
+    ctx match {
+      case c if c.int_alias != null => ir.IntegerType()
```
Comment (sundarshankar89) [inline core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/DataTypeBuilder.scala:12]: Snowflake has this weird challenge with datatypes, so I have seen data loss by mapping them to IntegerType directly. We should default to Number(38,0)  for all and, post-process, tackle this.

https://docs.snowflake.com/en/sql-reference/data-types-numeric#int-integer-bigint-smallint-tinyint-byteint

File: core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeCommandBuilder.scala [hunk 3]
```diff
@@ -0,0 +1,78 @@
+package com.databricks.labs.remorph.parsers.snowflake
+
+import com.databricks.labs.remorph.parsers.{IncompleteParser, ParserCommon, intermediate => ir}
+import SnowflakeParser.{StringContext => StrContext, _}
+
+import scala.collection.JavaConverters._
+class SnowflakeCommandBuilder
```
Comment (nfx) [inline core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeCommandBuilder.scala:7]:
```suggestion
class SnowflakeDDLBuilder
```

**Review by nfx — APPROVED** [review]
please address the comment from @sundarshankar89

**Comment by vil1** [comment]
@nfx any idea why codecov seems to not be triggered on my PRs?

**Pull request merged**